### PR TITLE
feat: Improve DdlRel message

### PR DESCRIPTION
**Pull request by moshe-yakovson**
- make Plan as a root of DdlRel
Plan => PlanRel  => Rel => DdlRel
then we can have necessary fields from Plan object
  Version, SimpleExtensionURI, SimpleExtensionDeclaration 

- add extension field to DdlRel to save table metadata

- add primary_key field to DdlRel message

@DaichikAmir @reutev

### CODE REVIEW

File: proto/substrait/algebra.proto [hunk 1]
```diff
@@ -410,6 +412,9 @@ message DdlRel {
   // The body of the CREATE VIEW
   Rel view_definition = 7;
 
+  // Field that can be used to extend/introduce any target spesific metadata
```
Comment (DaichikAmir) [inline proto/substrait/algebra.proto:415]: Specific (typo)

**Review by DaichikAmir — CHANGES_REQUESTED** [review]

**Comment by DaichikAmir** [comment]
Looks fine but I think that we can add fields that will probably be accepted , such as primary key

File: proto/substrait/algebra.proto [hunk 2]
```diff
@@ -410,6 +412,14 @@ message DdlRel {
   // The body of the CREATE VIEW
   Rel view_definition = 7;
 
+  // Field that can be used to extend/introduce 
+  // any target spesific metadata about the table data
```
Comment (moshe-yakovson) [inline proto/substrait/algebra.proto:416]:
```suggestion
  // any target specific metadata about the table data
```

**Review by DaichikAmir — APPROVED** [review]
Looks good to me , @reutev what do you think?

File: proto/substrait/algebra.proto [hunk 3]
```diff
@@ -410,6 +412,14 @@ message DdlRel {
   // The body of the CREATE VIEW
   Rel view_definition = 7;
 
+  // Field that can be used to extend/introduce 
+  // any target specific metadata about the table data
+  // like replication, partition etc
```
Comment (shaharv) [inline proto/substrait/algebra.proto:417]:
```suggestion
  // Field that could be used to extend or introduce any target specific metadata about the table,
  // like replication, partition, etc.
```

File: proto/substrait/algebra.proto [hunk 4]
```diff
@@ -410,6 +412,14 @@ message DdlRel {
...
+  ExtensionObject table_metadata = 8;
+
+  // Indexes of all primary keys in the table
```
Comment (shaharv) [inline proto/substrait/algebra.proto:420]:
```suggestion
  // Indices of all primary keys in the table
```

File: proto/substrait/algebra.proto [hunk 5]
```diff
@@ -410,6 +412,14 @@ message DdlRel {
...
+  repeated int32 primary_key = 9;
```
Comment (shaharv) [inline proto/substrait/algebra.proto:421]:
```suggestion
  repeated int32 primary_keys = 9;
```

**Pull request merged**